### ISSUE TITLE
-u argument to ansible-playbook is not properly handled

### ISSUE THREAD

**Issue by ktbyers** [napalm-automation/napalm-ansible#161]
Fix looks to be:

```diff
$ git diff ./napalm.py
diff --git a/napalm_ansible/plugins/action/napalm.py b/napalm_ansible/plugins/action/napalm.py
index ac37522..bbd70e0 100644
--- a/napalm_ansible/plugins/action/napalm.py
+++ b/napalm_ansible/plugins/action/napalm.py
@@ -13,7 +13,7 @@ class ActionModule(_ActionModule):
             provider = self._task.args.get('provider', {})
 
             provider['hostname'] = provider.get('hostname', provider.get('host', pc.remote_addr))
-            provider['username'] = provider.get('username', pc.connection_user)
+            provider['username'] = provider.get('username', pc.remote_user)
             provider['password'] = provider.get('password', pc.password)
             # Timeout can't be passed via command-line as Ansible defaults to a 10 second timeout
             provider['timeout'] = provider.get('timeout', 60)
```

**Comment by ktbyers** [comment]
Tested in Ansible 2.9rc3.

Haven't regression tested in other Ansible versions

**Comment by ktbyers** [comment]
Fixed in #162

**Comment by targuan** [comment]
Hello,

According to [ansible source code](https://github.com/ansible/ansible/blob/devel/lib/ansible/playbook/play_context.py#L321), the value of remote_user is overwritten by the current user if the connection local is set which used to be mandatory to use napalm.

Could you elaborate the context were the current action plugin is not working ?
I have been using it for a few years now without any issue.
But looking at the the play_context source code, there may be an issue when connection is set to another value than local (network for instance but I don't know if napalm module would work with this one). We may want to use connection_user if available (connection: local) if set and fallback to remote_user if not.
